pkg/kubelet/pleg/generic.go

```diff
@@ -116,40 +158,190 @@ func (g *GenericPLEG) relist() {
 	}()
 
 	// Get all the pods.
-	pods, err := g.runtime.GetPods(true)
+	podList, err := g.runtime.GetPods(true)
 	if err != nil {
 		glog.Errorf("GenericPLEG: Unable to retrieve pods: %v", err)
 		return
 	}
+	pods := kubecontainer.Pods(podList)
+
+	eventsByPodID := map[types.UID][]*PodLifecycleEvent{}
+	// Process all currently visible pods.
+	for _, pod := range pods {
+		g.podRecords.setCurrent(pod)
+		// Locate the old pod.
+		oldPod := g.podRecords.getOld(pod.ID)
+
+		// Process all currently visible containers in the pod.
+		for _, container := range pod.Containers {
```

Comment:
I feel it's cleaner to move the two for loop to a function which takes (oldPod, newPod)